pyproject.toml

```
@@ -3,5 +3,5 @@ requires = [
     "setuptools",
     "versioneer", 
     "wheel",
-    "oldest-supported-numpy"
+    "numpy>=2.0.0rc1"
```

For completeness, this would avoid having to drop Python 3.7 and 3.8
I personally agree that dropping them would probably be fine by now, but I just wanted to give a thorough picture of what's possible
```suggestion
    "oldest-supported-numpy ; python_version < '3.9'",
    "numpy>=2.0.0rc1 ; python_version >= '3.9'",
```

---

I think it's a better idea to keep compatibility where we can. I'd prefer this version of the build deps.